### DISCUSSION TITLE
Disconnected Deepgram socket after 20 seconds

### DISCUSSION THREAD

**Discussion by santoshchandak** [deepgram/community#969] [General help]
Dear Deepgram Support Team,

I hope you are doing well. We are encountering a critical issue in our production environment where the transcription stops recording after 20 seconds. This problem is affecting many of our users, and it is impacting the functionality of our software, which relies on Deepgram for recording transcripts. This occurred at 11:06 am EST.17th or 18th Oct

Could you please help us resolve this issue as soon as possible? Your prompt assistance would be greatly appreciated as it is affecting multiple users.

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The programming language you are working in (e.g. JavaScript, Python).
- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Comment by jkroll-deepgram** [comment]
Hi @santoshchandak, are you able to provide affected Deepgram request IDs, or your project ID?

When you say that "transcription stops recording," what results exactly are you seeing? Is there an error message, a blank transcript, or no reply from Deepgram at all?

Deepgram will close websockets after 12 seconds of inactivity, and you can send [KeepAlive messages](https://developers.deepgram.com/docs/audio-keep-alive) to keep the connection open.

We do not have a threshold to close a connection specifically at 20 seconds. Is it possible that you may have stopped streaming audio to Deepgram after a period of time, which would result in Deepgram not having any further transcription results?

**Comment by santoshchandak** [comment]
Hello @jkroll-deepgram 
This issue doesn't usually occur regularly in production. However, we encountered it a few days ago, and here are the details.

Project ID: 79a18085-9e83-4cde-b5cc-bddf13c43d89

User1:
Date: 18-10-2024
Time: 9:30 am - 12:30 pm EST (recording stopped 20 seconds into each session); 3:25 pm EST (transcription lag)
Device: MacBook Air, used with Google Chrome and Safari

User2:
Date: 18-10-2024
Time: 11:06 am EST (recording stopped); 11:16 am EST (transcription started with delay)
Device: Not specified

Also happened for some users on 19-10-2024

Getting blank transcript in deepgram response.
We don't have any threshold to stop streaming audio to Deepgram after a period of time.

**Comment by jkroll-deepgram** [comment]
Hi @santoshchandak, thanks for the additional information.

If you are still receiving streaming results back from Deepgram, but the transcription contents are are blank, that typically means that you are sending audio that Deepgram is not detecting speech in. The audio could be corrupted, or it could contain unintelligible speech or no speech.

Are you able to access the audio from the affected users on your side?

Going forward, I recommend that you log Deepgram request IDs (`response["metadata"]["request_id"]`) within your application, so that when an issue arises, we can look up the logs and details of the exact requests that were affected.
